web/src/main/java/org/springframework/security/web/firewall/StrictHttpFirewall.java

```diff
@@ -355,6 +386,79 @@ private void urlBlocklistsRemoveAll(Collection<String> values) {
 		this.decodedUrlBlocklist.removeAll(values);
 	}
 
+	/**
+	 * <p>
+	 * Determines which header names should be allowed.
+	 * The default is to reject header names that contain ISO control characters
+	 * and characters that are not defined.
+	 * </p>
+	 *
+	 * @param allowedHeaderNames the predicate for testing header names
+	 * @see Character#isISOControl(int)
+	 * @see Character#isDefined(int)
+	 * @since 5.4
+	 */
+	public void setAllowedHeaderNames(Predicate<String> allowedHeaderNames) {
```

Comment:
Will you please move these so that they are next to the other public setters?

Reply:
Done :+1: